oauth-common/src/main/java/io/strimzi/kafka/oauth/services/OAuthMetrics.java

```diff
@@ -90,26 +111,48 @@ private Metrics initKafkaMetrics() {
 
     private List<MetricsReporter> initReporters() {
         AbstractConfig kafkaConfig = initKafkaConfig();
-        List<MetricsReporter> reporters = kafkaConfig.getConfiguredInstances(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
-                MetricsReporter.class);
-
-        JmxReporter reporter = new JmxReporter();
-        reporter.configure(configMap);
-
-        reporters.add(reporter);
+        List<MetricsReporter> reporters = new ArrayList<>(3);
+        if (configMap.get(STRIMZI_METRIC_REPORTERS) != null) {
+            reporters = kafkaConfig.getConfiguredInstances(STRIMZI_METRIC_REPORTERS, MetricsReporter.class);
+        } else {
+            log.warn("Configuration option '{}' is not set, OAuth metrics will not be exported to JMX", STRIMZI_METRIC_REPORTERS);
+        }
         return reporters;
     }
 
     private AbstractConfig initKafkaConfig() {
-        ConfigDef configDef = new ConfigDef()
-                .define(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
-                        ConfigDef.Type.LIST,
-                        Collections.emptyList(),
-                        new ConfigDef.NonNullValidator(),
-                        ConfigDef.Importance.LOW,
-                        CommonClientConfigs.METRIC_REPORTER_CLASSES_DOC);
-
-        return new AbstractConfig(configDef, configMap);
+        ConfigDef configDef = addMetricReporterToConfigDef(new ConfigDef(), STRIMZI_METRIC_REPORTERS);
+        return new AbstractConfig(configDef, toMapOfStringValues(configMap));
+    }
+
+    private ConfigDef addMetricReporterToConfigDef(ConfigDef configDef, String name) {
+        return configDef.define(name,
+                ConfigDef.Type.LIST,
+                Collections.emptyList(),
+                new ConfigDef.NonNullValidator(),
+                ConfigDef.Importance.LOW,
+                CommonClientConfigs.METRIC_REPORTER_CLASSES_DOC);
+    }
+
+    private Map<String, String> toMapOfStringValues(Map<String, ?> configMap) {
+        HashMap<String, String> result = new HashMap<>();
+        for (Map.Entry<String, ?> ent: configMap.entrySet()) {
+            Object val = ent.getValue();
+            if (val == null) {
+                continue;
+            }
+            if (val instanceof Class) {
+                result.put(ent.getKey(), ((Class<?>) val).getCanonicalName());
+            } else if (val instanceof List) {
+                String stringVal = ((List<?>) val).stream().map(String::valueOf).collect(Collectors.joining(","));
+                if (!"".equals(stringVal)) {
```

Comment:
Can we use `isEmpty()`?

Reply:
In this case we can as stringVal cannot be null here.